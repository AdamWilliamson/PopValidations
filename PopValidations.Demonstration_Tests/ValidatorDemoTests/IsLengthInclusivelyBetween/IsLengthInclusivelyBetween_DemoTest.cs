using ApprovalTests;
using Newtonsoft.Json.Linq;
using PopValidations.Swashbuckle_Tests.Helpers;
using PopValidations_Tests.TestHelpers;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using Xunit;

namespace PopValidations.Demonstration_Tests.ValidatorDemoTests.IsLengthInclusivelyBetween;

public static class IsLengthInclusivelyBetween
{
    public record InputObject(string? NString, List<int> Array);

    public class Validator : AbstractValidator<InputObject>
    {
        public Validator()
        {
            Describe(x => x.NString).IsLengthInclusivelyBetween(1,5);
            Describe(x => x.Array).IsLengthInclusivelyBetween(1, 5);
        }
    }

    public class TestController : ControllerBase<InputObject> { }
}

public class IsLengthInclusivelyBetween_DemoTest
{
    [Fact]
    public async Task Validation()
    {
        // Arrange
        var validationRunner = ValidationRunnerHelper.BasicRunnerSetup(new IsLengthInclusivelyBetween.Validator());

        // Act
        var validationResult = await validationRunner.Validate(
            new IsLengthInclusivelyBetween.InputObject(NString: "6Chars", Array: new() { })
        );

        // Assert
        Approvals.VerifyJson(JsonConverter.ToJson(validationResult));
    }

    [Fact]
    public void Description()
    {
        // Arrange
        var descriptionRunner = ValidationRunnerHelper.BasicRunnerSetup(new IsLengthInclusivelyBetween.Validator());

        // Act
        var descriptionResult = descriptionRunner.Describe();

        // Assert
        Approvals.VerifyJson(JsonConverter.ToJson(descriptionResult));
    }

    [Fact]
    public async Task OpenApi()
    {
        // Arrange
        var config = new WebApiConfig();
        var setup = new TestSetup<
               IsLengthInclusivelyBetween.TestController,
               IsLengthInclusivelyBetween.Validator,
               IsLengthInclusivelyBetween.InputObject
           >();

        // Act
        var helper = await setup.GetHelper(config);
        JObject json = JObject.Parse(helper.Content);
        var match = json["components"]["schemas"].FirstOrDefault();

        // Assert
        Approvals.VerifyJson(match?.ToString(Newtonsoft.Json.Formatting.None));
    }
}