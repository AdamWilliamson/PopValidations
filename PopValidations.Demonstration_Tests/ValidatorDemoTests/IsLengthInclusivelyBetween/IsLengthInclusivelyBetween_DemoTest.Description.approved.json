{
    "results": [
        {
            "property": "NString",
            "outcomes": [
                {
                    "validator": "IsLengthInclusivelyBetweenValidation",
                    "message": "Must be between 1 and 5 inclusive.",
                    "values": [
                        {
                            "key": "startValue",
                            "value": "1"
                        },
                        {
                            "key": "endValue",
                            "value": "5"
                        }
                    ]
                }
            ],
            "validationGroups": []
        },
        {
            "property": "Array",
            "outcomes": [
                {
                    "validator": "IsLengthInclusivelyBetweenValidation",
                    "message": "Must be between 1 and 5 inclusive.",
                    "values": [
                        {
                            "key": "startValue",
                            "value": "1"
                        },
                        {
                            "key": "endValue",
                            "value": "5"
                        }
                    ]
                }
            ],
            "validationGroups": []
        }
    ]
}